source/fe/mapping_cartesian.cc

```diff
@@ -1166,32 +1168,16 @@ MappingCartesian<dim, spacedim>::transform_unit_to_real_cell(
   const Point<dim>                                           &p) const
 {
   Assert(is_cartesian(cell), ExcCellNotCartesian());
+  if (dim != spacedim)
+    DEAL_II_NOT_IMPLEMENTED();
```

Comment:
This does not output the controlling condition `dim==spacedim`. Would you mind writing this as
```
  Assert (dim==spacedim,
               ExcNotImplemented());
```
? Same below.